torchgeo/datasets/mdas.py

```diff
@@ -0,0 +1,347 @@
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new MDAS dataset instance.
+
+        Args:
+            root: Root directory where the dataset should be stored.
+            subareas: The subareas to load. Options are 'sub_area_1', 'sub_area_2', 'sub_area_3'.
+            modalities: The modalities to load. Options are '3K_DSM', '3K_RGB', 'HySpex', 'EeteS_EnMAP_10m', 'EeteS_EnMAP_30m', 'EeteS_Sentinel_2_10m', 'Sentinel-2', 'Sentinel-1', 'OSM_label'.
+            transforms: A function/transform that takes in a dictionary and returns a transformed version.
+            download: if True, download dataset and store it in the root directory
+            checksum: If True, check the integrity of the dataset after download.
+
+        Raises:
+            AssertionError: If the subareas or modalities are not valid.
+            DatasetNotFoundError: If the dataset is not found.
+        """
+        self.root = root
+        self.download = download
+        assert all(
+            sub in self.valid_subareas for sub in subareas
+        ), f'Subareas must be one of {self.valid_subareas}'
+        self.subareas = subareas
+        assert all(
+            mod in self.valid_modalities for mod in modalities
+        ), f'Modalities must be one of {self.valid_modalities}'
+        self.modalities = modalities
+        self.transforms = transforms
+        self.checksum = checksum
+
+        self._verify()
+        self.files = self._load_files()
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset."""
+        return len(self.files)
+
+    def _load_files(self) -> list[dict[str, str]]:
+        """Return the paths of the files in the dataset."""
+        files = []
+        for subarea in self.subareas:
+            subarea_files = {}
+            for modality in self.modalities:
+                subarea_files[modality] = os.path.join(
+                    self.root,
+                    self.ds_root_name,
+                    subarea,
+                    f'{modality}_{self._format_subarea(subarea)}.tif',
+                )
+            files.append(subarea_files)
+        return files
+
+    def _format_subarea(self, subarea: str) -> str:
+        """Format the subarea name.
+
+        Args:
+            subarea: The subarea string to format.
+
+        Returns:
+            formatted subarea string for files
+        """
+        parts = subarea.split('_')
+        return parts[0] + '_' + parts[1] + parts[2]
+
+    def _load_image(self, path: Path) -> Tensor:
+        """Load an image from a given path."""
+        with rio.open(path) as src:
+            img = src.read()
+            if img.dtype == np.uint16:
+                img = img.astype(np.int32)
+            if 'osm_landuse' in str(path):
+                img = np.vectorize(self.landuse_mapping.get)(img)
+
+            return torch.from_numpy(img)
+
+    def __getitem__(self, idx: int) -> dict[str, Any]:
```

Comment:
```suggestion
    def __getitem__(self, idx: int) -> dict[str, Tensor]:
```
I think we only return Tensors, right?